sdk/core/azure-core/src/main/java/com/azure/core/models/spatial/Point.java

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.models.spatial;
+
+import com.azure.core.annotation.Immutable;
+
+/**
+ * Represents a location in (x, y) coordinate space.
+ */
+@Immutable
+public final class Point {
+
+    /*
+     * The x-axis point coordinate.
+     */
+    private final float xCoordinate;
+
+    /*
+     * The y-axis point coordinate.
+     */
+    private final float yCoordinate;
+
+    /**
+     * Constructs a Point object.
+     *
+     * @param xCoordinate The x-axis point coordinate.
+     * @param yCoordinate The y-axis point coordinate.
+     */
+    public Point(final float xCoordinate, final float yCoordinate) {
+        this.xCoordinate = xCoordinate;
+        this.yCoordinate = yCoordinate;
+    }
+
+    /**
+     * Gets the x-coordinate value.
+     *
+     * @return The x-axis coordinate value.
+     */
+    public float getX() {
+        return this.xCoordinate;
+    }
+
+    /**
+     * Gets the y-coordinate value.
+     *
+     * @return The y-axis coordinate value.
+     */
+    public float getY() {
+        return this.yCoordinate;
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) return true;
```

Comment:
I thought it was against our coding style guidelines to use conditionals without `{}`